docs/design/collaboration/Building the collaboration paradigm between App Developers and Platform Developers with Kusion.md

```diff
@@ -0,0 +1,235 @@
+
+### Platform developer's view
+Humanitec provides many [built-in drivers](https://developer.humanitec.com/integration-and-extensions/drivers/overview/#built-in-drivers), if existing drivers can not meet your requirements, you can build your own custom drivers. In general, Humanitec Drivers are similar to Terraform providers which is an API service fulfilling the Humanitec Driver Interface. According to the official website, they are working on a new API spec, so you can only contact their staff to build your driver now, but you can leverage Terraform providers with its [generic Terraform drivers](https://developer.humanitec.com/integration-and-extensions/drivers/generic-drivers/terraform/).
+Platform engineers build the driver and they decide the driver's inputs. They also are responsible for configuring matching criteria to provide a resource in a given resource context.
+
+### Application developer's view
+
+The  Score specification consists of four parts: workload definition, containers, resources, and service. All attributes application developers can configure are defined in this [specification](https://docs.score.dev/docs/reference/score-spec-reference/). Taking Postgres as an example, app developers only need to declare that they need a Postgres database in the resources block and all the other attributes like engine version, storage size, etc. are configured by the platform developers.
+
+```yaml
+apiVersion: score.dev/v1b1
+metadata:
+  name: backend
+
+containers:
+... ...
+
+resources:
+  db:
+    type: postgres
+```
+
+### Pros
+
+- The score file significantly reduces app developers' cognitive load and separates app developers' and platform developers' concerns.
+- Support the complete software lifecycle, including monitoring, logging, secret management, and more.
+
+### Cons
+
+- Most of the work was handed over to platform developers. Different applications may have different requirements for the same resource attribute, and platform developers' workload has increased several times.
+It is reasonable to allow app developers to configure certain attributes, such as the database engine version since it can impact their business logic code.
+
+## Summary
+
+We won't list all the projects we have researched here. Each project has its advantages and we have learned a lot from them, but none of them can meet the goals defined above. The following section will introduce our proposal
+
+# Our Design
+
+According to our experiences and extensive conversations with other companies, teams included in SLCD(Software Life Cycle Development) are generally divided into two parts: platform engineers and application engineers. 
```

Comment:
Based on our large-scale practical experience and extensive industry research, teams participate in SDLC(Software Development Life Cycle) are generally divided into two roles: